lib/spack/spack/test/concretization/core.py

```diff
@@ -1719,27 +1720,27 @@ def test_reuse_with_unknown_namespace_dont_raise(
 
     @pytest.mark.regression("45538")
     def test_reuse_from_other_namespace_no_raise(self, tmpdir, temporary_store, monkeypatch):
-        myrepo = spack.repo.MockRepositoryBuilder(tmpdir, namespace="mock_repo")
+        myrepo = spack.test.conftest.MockRepositoryBuilder(tmpdir)
         myrepo.add_package("zlib")
 
         builtin = spack.concretize.concretize_one("zlib")
         PackageInstaller([builtin.package], fake=True, explicit=True).install()
 
         with spack.repo.use_repositories(myrepo.root, override=False):
             with spack.config.override("concretizer:reuse", True):
-                myrepo = spack.concretize.concretize_one("mock_repo.zlib")
+                zlib = spack.concretize.concretize_one(f"{myrepo.namespace}.zlib")
 
-        assert myrepo.namespace == "mock_repo"
+        assert zlib.namespace == myrepo.namespace
 
     @pytest.mark.regression("28259")
     def test_reuse_with_unknown_package_dont_raise(self, tmpdir, temporary_store, monkeypatch):
-        builder = spack.repo.MockRepositoryBuilder(str(tmpdir), namespace="myrepo")
+        builder = spack.test.conftest.MockRepositoryBuilder(str(tmpdir))
```

Comment:
In this and the other test above, tmpdir is only used by MockRepositoryBuilder. I think things would be simplified if you put another fixture in conftest like this:

```python
@pytest.fixture
def repo_builder(tmpdir):
    builder = MockRepositoryBuilder(str(tmpdir))
    try:
        yield builder
    finally:
        # delete any packages / namespaces added to sys.modules with builder
```

Then you could just require the repo_builder fixture here and remove the need for the tests to import directly from conftest and set the builder up.

Reply:
The upside of constructing MockRepository the normal way is typing; fixtures have type erasure

Reply:
You can add types for fixtures:

```
def test_foo(repo_builder: MockRepositoryBuilder):
    …
```

Reply:
yes, but you wanted

> remove the need for the tests to import directly from conftest